src/testing/test_core.py

```diff
@@ -0,0 +1,122 @@
+"""
+Test module: test_core.py
+
+Testing module focuses on testing the core functions of the analysis pipeline.
+"""
+
+import unittest
+from typing import Iterator
+
+from PIL import Image
+
+from src.common.file_io import load_tiff_images
+from src.structs.selections import ImageCropSelection
+from src.utils.image_utils import image_crop_walk
+
+
+class TestUtils(unittest.TestCase):
+    """Tests utilities functions"""
+
+    def test_load_tiff_img(self) -> None:
+        """With a given path to a `.tiff` file, should return and Image.Image
+        object"""
+
+        # loading directory
+        loaded_img = load_tiff_images("./test_data/tiff_images")
+
+        # checking if iterator is returned
+        self.assertIsInstance(loaded_img, Iterator)
+
+        # checking if elements are loaded as Image.Image types
+        images = list(loaded_img)
+        img_sel = images[0]
+        self.assertIsInstance(img_sel, Image.Image)
+
+    def test_load_tiff_with_empty_dir(self) -> None:
+        """Testing loader with an empty directory. Should expect a
+        `FileNotFoundError` if no `.tiff` files are found"""
+        # loading
+        jpeg_path = "./test_data/empty_dir"
```

Comment:
I'm assuming these paths are ok as strings instead of `pathlib.Path()` objects?

Reply:
The `load_tiff_images` function take's in both `str` and `pathlib.Path()` objects. If `str` are passed, they are automatically converted into `pathlib.Path()` objects:

https://github.com/axiomcura/macrophage-striation-analysis/blob/5c8727d3cf1cd134543c99b1841460075e86e67e/src/common/file_io.py#L44-L49

But I should make another test that uses a `pathlib.Path()` object Thanks!

Reply:
Test has been added!